homepage/scripts/scripts.js

```diff
@@ -269,6 +271,16 @@ function loadStyles() {
     if (signedInCookie && isSignedInUser && !window.location.href.includes('/fragments/')) {
       window.location.reload();
     }
-  })
+  });
   await loadAreaPromise;
+
+  // for media_ after loading loadAreaPromise
+  const observeCallback = (mutationList, observer) => {
+    for (const mutation of mutationList) {
+      replaceDotMedia(mutation.target);
+    }
+  };
+  const observer = new MutationObserver(observeCallback);
+  observer.observe(document.querySelector('header'), { childList: true, subtree: true });
+  observer.observe(document.querySelector('footer'), { childList: true, subtree: true });
```

Comment:
I'd still have the concern any block with media would be broken if it loads delayed, no?
You might be stuck with observing the whole document, in which case you might want to check if that has any performance impact